core/src/main/java/hudson/model/Item.java

```diff
@@ -134,24 +138,28 @@
      * Gets the relative name to this item from the specified group.
      *
      * @param g
-     *      The ItemGroup instance used as context to evaluate the relative name of this AbstractItem
+     *      The {@link ItemGroup} instance used as context to evaluate the relative name of this item
      * @return
-     *      The name of the current item, relative to p. Nested ItemGroups are separated by {@code /} character.
+     *      The name of the current item, relative to p. Nested {@link ItemGroup}s are separated by {@code /} character.
      * @since 1.419
      * @return
      *      String like "../foo/bar".
-     *      {@code null} if item parents is not an {@link ItemGroup}.
+     *      {@code null} if one of item parents is not an {@link Item}.
      */
-    default String getRelativeNameFrom(ItemGroup g) {
+    @Nullable
```

Comment:
Not actually sure why these are all `@Nullable` rather than `@CheckForNull`. Is there some circumstance where a caller would statically know the return value cannot be null?

Reply:
Yes, generally "in all sane cases" assuming you do not resolve names for things like Promoted Builds plugin or pass nulls. I can add `@CheckForNull`, but it may become just another `Jenkins.getInstance()`